### PR TITLE
BAEL-1069: Guide to diamond operator in Java

### PR DESCRIPTION
**Pull request by Sgitario**

### CODE REVIEW

File: core-java/src/test/java/org/baeldung/java/diamond/DiamondOperatorTest.java [hunk 1]
```diff
@@ -0,0 +1,13 @@
+package org.baeldung.java.diamond;
+
+import static org.junit.Assert.assertNotNull;
+
+import org.junit.Test;
+
+public class DiamondOperatorTest {
```
Comment (KevinGilmore) [inline core-java/src/test/java/org/baeldung/java/diamond/DiamondOperatorTest.java]: rename to DiamondOperatorUnitTest

File: core-java/src/test/java/org/baeldung/java/diamond/DiamondOperatorTest.java [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+    @Test
+    public void shouldCreateCarUsingDiamondWithoutWarnings() {
```
Comment (KevinGilmore) [inline core-java/src/test/java/org/baeldung/java/diamond/DiamondOperatorTest.java]: Not a very interesting test...but let's at least follow the naming convention for tests: givenX_whenY_thenZ (the givenX is optional). So you would have something like: whenCreateCarUsingDiamondOperator_thenSuccess

File: core-java/src/test/java/org/baeldung/java/diamond/DiamondOperatorTest.java (hunk 1 above)
Comment (Sgitario) [reply core-java/src/test/java/org/baeldung/java/diamond/DiamondOperatorTest.java]: Done

File: core-java/src/test/java/org/baeldung/java/diamond/DiamondOperatorTest.java (hunk 2 above)
Comment (Sgitario) [reply core-java/src/test/java/org/baeldung/java/diamond/DiamondOperatorTest.java]: Renamed. I know it's not a very interesting test, but taking into account the article...

Comment (KevinGilmore) [reply core-java/src/test/java/org/baeldung/java/diamond/DiamondOperatorTest.java]: That's fair.

**Pull request merged**